i3pystatus/core/io.py

```diff
@@ -50,24 +52,37 @@ class StandaloneIO(IOHandler):
     and the i3bar protocol header
     """
 
+    __is_click_event = False
+
     n = -1
     proto = [
         {"version": 1, "click_events": True}, "[", "[]", ",[]",
     ]
 
-    def __init__(self, click_events, interval=1):
+    def __init__(self, click_events, modules, interval=1):
         super().__init__()
         self.interval = interval
         self.proto[0]['click_events'] = click_events
         self.proto[0] = json.dumps(self.proto[0])
+        self.modules = modules
 
     def read(self):
         while True:
+            received_signal = None
             try:
-                time.sleep(self.interval)
+                received_signal = signal.sigtimedwait([signal.SIGUSR1], self.interval)
+            except InterruptedError:
+                logging.getLogger("i3pystatus").exception("Interrupted system call:")
             except KeyboardInterrupt:
                 return
 
+            if received_signal:
+                if StandaloneIO.__is_click_event:
+                    StandaloneIO.__is_click_event = False
```

Comment:
Uh, not sure if I get this right.

_i_c_e is set (at class construction) to True, if it's true && received_signal, it's set to false. I don't quite grasp this - why is it necessary?